esp-hal-common/src/rtc/esp32c2.rs

```diff
@@ -59,10 +59,16 @@ pub(crate) fn init() {
 }
 
 pub(crate) fn configure_clock() {
+    #[cfg(feature = "esp32c2_40mhz")]
     assert!(matches!(
         RtcClock::get_xtal_freq(),
         XtalClock::RtcXtalFreq40M
     ));
+    #[cfg(feature = "esp32c2_26mhz")]
+    assert!(
+        matches!(RtcClock::get_xtal_freq(), XtalClock::RtcXtalFreq26M),
```

Comment:
Is `RtcClock::get_xtal_freq()` reliable? Would be possible to automatically figure out the crystal clock and change it in configure instead of the featues?

Reply:
From what I have seen it is not - like when flashing a 40MHz bootloader it will tell us 40MHz

Reply:
Also, in ESP-IDF it's needed to use `menuconfig` to configure it - so I assume there is no reliable way to infer it
But I'd certainly prefer to not need those features